src/wms/src/main/java/org/geoserver/wms/WMSRequests.java

```diff
@@ -434,6 +437,30 @@ private static String getLayerStyles(Layer[] layers) {
         return params;
     }
 
+    @SuppressWarnings("unchecked")
+    private static int getRawLayerIndex(GetMapRequest req, int layerIndex) {
+        List<String> names = KvpUtils.readFlat(req.getRawKvp().get("layers"));
+        if (names.size() != req.getLayers().size()) {
+            // Finds the index in the raw layers KVP parameter that corresponds
```

Comment:
This should be the javadoc of this method. Even if not public, it does not hurt to see what the method does without checking its code. Maybe expand a bit the explanation, I know what WMS does but I still had to open the code in the IDE to figure out what was going on.

Reply:
I added javadoc and more comments to this method.